manifest.go

```diff
@@ -67,8 +68,20 @@ func validateManifest(s string) ([]error, error) {
 					for key, value := range v.(map[string]interface{}) {
 						// Check if the key is valid
 						switch key {
-						case "name", "branch", "revision", "version", "source":
+						case "name", "branch", "version", "source":
 							// valid key
+						case "revision":
+							if valueStr, ok := value.(string); ok {
+								// Check if sha1 hash is abbreviated
+								validGit := regexp.MustCompile("[a-f0-9]{40}").MatchString(valueStr)
```

Comment:
I would recommend renaming this variable to `validHash` instead, since it could be from git or hg